File: java-akka/src/main/java/org/spiral/akka/java/demo/LoanRateQuotes.java
package org.spiral.akka.java.demo;

import akka.actor.AbstractActor;
import akka.actor.OneForOneStrategy;
import akka.actor.SupervisorStrategy;
import akka.japi.Function;
import scala.Tuple2;
import scala.collection.Iterable;
import scala.collection.mutable.ArraySeq;
import scala.concurrent.duration.Duration;

import java.util.concurrent.TimeUnit;

/**
 * 单项单款利率报价实例
 *
 * @author dengguoqing
 * @date 2019/5/21
 */
public class LoanRateQuotes extends AbstractActor {
    @Override
    public Receive createReceive() {
        return null;
    }

    @Override
    public SupervisorStrategy supervisorStrategy() {
        Iterable<Tuple2<Exception, SupervisorStrategy.Directive>> list = new ArraySeq<>(
                10);
        Function<Throwable, SupervisorStrategy.Directive> function = (e) -> {
            if (e instanceof Exception) {
                return SupervisorStrategy.resume();
            }
            return null;
        };
        SupervisorStrategy.makeDecider(function);
        return new OneForOneStrategy(5, Duration.create(1,
                                                        TimeUnit.MINUTES),
                                     true,
                                     SupervisorStrategy.defaultDecider());
    }
}

File: scala-akka/src/main/java/org/spiral/akka/scala/demo/LoanRateQuotes.scala
package org.spiral.akka.scala.demo

import java.util.concurrent.TimeUnit

import akka.actor.SupervisorStrategy.{Escalate, Restart, Resume, Stop}
import akka.actor.{Actor, OneForOneStrategy, SupervisorStrategy}

import scala.concurrent.duration.Duration

/**
  * ${DESCRIPTION}
  *
  * @author dengguoqing
  * @since 2019/5/21
  */
class LoanRateQuotes extends Actor {


  override def supervisorStrategy: SupervisorStrategy = {
    OneForOneStrategy(maxNrOfRetries = 5,
      withinTimeRange = Duration.apply(1, TimeUnit.MINUTES)) {
      case NullPointerException => Restart
      case ArithmeticException => Resume
      case IllegalArgumentException => Stop
      case UnsupportedOperationException => Stop
      case Exception => Escalate
    }
  }

  override def receive: Receive = {
    return null
  }
}
